File: exercises/03 PhotoSearcher/03 PhotoSearcher/PhotoDeleter.cs
using System;

namespace _03_PhotoSearcher
{
    public class PhotoDeleter
    {
        private IPhotoRepository photoRepository;

        public PhotoDeleter(IPhotoRepository photoRepository)
        {
            this.photoRepository = photoRepository;
        }

        public void DeletePhotos(FuzzyDate fuzzyDate)
        {
            // Use test-driven development to implement this method
            throw new NotImplementedException();
        }

        public void DeletePhotos(FuzzyDate fuzzyDate, Color color, ContentType? contentType)
        {
            // Use test-driven development to implement this method
            throw new NotImplementedException();
        }

    }
}

File: exercises/03 PhotoSearcher/C++/03 PhotoSearcher/PhotoDeleter.cpp
#include "PhotoDeleter.h"
#include <vector>
#include <algorithm>
#include <iterator>
#include "FuzzyDate.h"
#include "Color.h"
#include "IPhotoRepository.h"
#include "IPhoto.h"

using namespace std;

PhotoDeleter::PhotoDeleter(IPhotoRepository* photoRepository) :
	_photoRepository(photoRepository)
{
}

PhotoDeleter::~PhotoDeleter()
{
}

void PhotoDeleter::deletePhotos(const FuzzyDate& fuzzyDate)
{
}

void PhotoDeleter::deletePhotos(const FuzzyDate & fuzzyDate, const Color & color, ContentType contentType)
{
}
